src/hclib-locality-graph.c

```diff
@@ -1141,8 +1141,8 @@ hclib_locale_t *hclib_get_closest_locale_of_types(hclib_locale_t *locale,
     int to_visit_index = 0;
     int *to_visit = (int *)malloc(sizeof(int) * n_locales);
     hclib_locale_t *curr = locale;
-    while (!contains(curr->type, locale_types, n_locale_types) &&
-            visiting_index <= n_locales) {
+    while (visiting_index <= n_locales &&
```

Comment:
Look like the out of bounds transferred to another array. When data is not found, then visiting_index is n_locales+1. So inside the loop, in line 1155, this would access to_visit[n_locales] which might cause segmentation fault since to_visit is only of size n_locales. Either increase size of to_visit by 1 or better make the loop only access up to the legal range of n_locales-1

Reply:
Good point. I've just pushed another commit that should address this problem with this function.